### PR TITLE
Molecule refactor testing

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
Refactored the molecule class to make use of the beadstructure as a base class, then I proceeded to remove duplicate functionality. 

The bead structure makes it possible to make comparisons between the structures of molecules. This way you can actually use the isStructureEquivalent function to compare molecules based on their structure. I have also attempted to add some testing.

### CODE REVIEW

File: src/tests/CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,7 @@ foreach(PROG
   test_interaction
   test_lammpsdatareader 
   test_lammpsdumpreaderwriter
+  test_molecule
```
Comment (junghans) [inline src/tests/CMakeLists.txt]: There is already a test called `test_molecule` in xtp, so you will need to rename one of the two.

Comment (JoshuaSBrown) [reply src/tests/CMakeLists.txt]: Good point any suggestions?

Comment (junghans) [reply src/tests/CMakeLists.txt]: `csg_test_molecule` ?

**Comment by JoshuaSBrown** [comment]
This breaks ctp.

**Comment by JoshuaSBrown** [comment]
This branch was put on hold because of an error that is addressed here: https://github.com/votca/csg/pull/283

will need to propagate the fix up to this branch.

**Comment by JoshuaSBrown** [comment]
This branch should not be merged with master at this time.

**Pull request closed without merging**